### ISSUE TITLE
envoyproxy/envoy-build-centos build is failed 

### ISSUE THREAD

**Issue by Alexcei88** [envoyproxy/envoy#18365]
Some time ago I was successful to build Envoy for centos. I was very happy.

`IMAGE_NAME=envoyproxy/envoy-build-centos ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.release.server_only'`

But now I catch follow

```
Use --sandbox_debug to see verbose messages from the sandbox
/build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/sandbox/processwrapper-sandbox/8/execroot/envoy/external/com_googlesource_chromium_v8/wee8 /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/sandbox/processwrapper-sandbox/8/execroot/envoy
ERROR at //testing/gmock/BUILD.gn:11:1: Dependency not allowed.
source_set("gmock") {
^--------------------
The item //testing/gmock:gmock
can not depend on //third_party/googletest:gtest_config
because it is not in //third_party/googletest:gtest_config's visibility list: [
  //third_party/googletest:*
]
```

Any idea how to fix it? I cannot catch what changes happened.

**Comment by phlax** [comment]
> Some time ago I was successful to build Envoy for centos. I was very happy.

you shouldnt require the centos build image to build *for* centos - i would suggest using the ubuntu build image

the centos build image is not properly supported (not sure exactly how that relates to the issue you have)

i have opened a ticket (here https://github.com/envoyproxy/envoy-build-tools/issues/149) to remove the centos build image altogether for the above reason

**Comment by Alexcei88** [comment]
But my target machine where I want to execute envoy is Centos 7.
So, I have problem with GLIBC version. We cannot refuse to build from Centos7.

**Comment by Alexcei88** [comment]
I agree that build doesn't happened for zero. But I could resolve all problems with build earlier. But for this problem I have no idea.

**Comment by phlax** [comment]
> So, I have problem with GLIBC version.

i see.

im not sure if the envoy repo itself will support a lesser version of GLIBC any more

also not sure how that relates to the issue you are facing

the one thing i am fairly sure of is that the centos build image is not properly maintained

https://github.com/envoyproxy/envoy-build-tools/commits/main/build_container/Dockerfile-centos

**Comment by bartsmykla** [comment]
I faced this issue today, and found out the bug was in v8, and here is the fix: https://github.com/v8/v8/commit/eac21d572e92a82f5656379bc90f8ecf1ff884fc

I would love to contribute with PR updating v8 to version with this patched (`9.5.164` - `9.6.152`). @PiotrSikora as you are the one, who bumped the v8 version, would you be so kind and assist me with advice if I can do it?

**Comment by phlax** [comment]
@bartsmykla iiuc the problem/resolution, you probs need to start by looking here https://github.com/envoyproxy/envoy/blob/34edbca8389191f65bb8dbbb00fb5d5b036535b6/bazel/repository_locations.bzl#L822-L835

**Comment by bartsmykla** [comment]
yes, I know, and am working on a PR, but my question was mostly if the community is ok with me doing so :-)

**Comment by Alexcei88** [comment]
Yep, I rolled back gn to old version(I took 1935) and it seems like envoy will be built. It's in progress...

**Comment by Alexcei88** [comment]
Rollback to older version gn inside docker image solved error and I got build for centos7.
I think issue can be closed. 

I made 3 changes inside docker image locally to build envoy:
* modify PATH according [this issue](https://github.com/envoyproxy/envoy/issues/12535)
* install gn according [issue](https://github.com/envoyproxy/envoy/issues/15119)
* install libstdc++ library `yum install glibc-static libstdc++-static`

I think it is not so many issues to fix docker build image.
Since I am very interested  to save build envoy for centos7, may be if I will have time I can try to fix docker build image for centos 7.
